[IV-24-25] Objetivo 1

lmchaves opened the pull request:
* [x] ¿Las historias de usuario y los milestones constituyen una guía
    suficiente para poder comenzar el desarrollo de un proyecto?
Sí, cada milestone establece qué funcionalidad se va a implementar, lo que permite empezar a trabajar desde la estructura del taller hasta la implementación de la lógica de negocio.
* [x] ¿Los milestones están ordenados correctamente, de forma que el 1 o
   0 sea lo primero que hay que empezar a publicar?
Sí, el Milestone 0 está relacionado con el modelado del dominio, que es el primer paso lógico. A partir de ahí, los milestones avanzan de forma secuencial y lógica.
* [x] ¿Todos los milestones se construyen sobre el anterior, teniendo un
   orden lógico el desarrollo?
Sí.
* [x] ¿El milestone 0 tiene asignada alguna historia de usuario?
 M0 establece las bases para poder implementar las historias de usuario
* [x] ¿El milestone 0 corresponde con lo mencionado varias veces en
   clase, y en concreto con lo que se solicita en el objetivo
   siguiente?
Sí, se enfoca en modelar el dominio del problema, lo que es fundamental antes de implementar cualquier lógica de negocio.
* [x] ¿Todos los milestones describen correctamente un producto y qué
   lo hace válido, y no un  concepto vago, una funcionalidad o una tarea?
Sí, se describen claramente los criterios que hacen que el producto sea válido.
* [x] ¿Los milestones son graduales, o hay un salto grande entre ellos?
Son graduales, ya que primero se modela, luego se consulta el inventario y después se gestiona la distribución de piezas entre sedes. Esto evita saltos grandes y permite un avance progresivo.
* [x] ¿Los milestones iniciales son internos, y hay a continuación milestones o
   productos externos?
Sí, el Milestone 0 es interno (modelado del sistema), mientras que los siguientes (M1 y M2) son externos, ya que afectan la interacción con usuarios del sistema, como mecánicos y el jefe del taller.
* [x] ¿Los milestones son mínimos, es decir, incluyen un producto
   mínimamente viable para poder avanzar?
Sí.
* [x] ¿El milestone 0 es mínimo *y* viable?
Sí.
* [x] ¿Se indica claramente cómo comprobar la viabilidad del PMV de cada hito? Es
    decir, ¿se dice qué requisitos técnicos tiene que cumplir el producto de
    cada hito?
Sí, cada milestone incluye criterios de viabilidad claros.
* [x] ¿Entre dos milestones consecutivos, no hay ningún PMV posible o por el
    contrario podrían desarrollarse muchas posibles etapas internas o externas
    de un proyecto?
* [x] ¿He tenido en cuenta en las historias de usuario que se trata de un
   proyecto que se desplegará en la nube?
Sí.
* [x] ¿Están descritos todos los conceptos mencionados en las HUs con
   suficiente claridad, generalmente en documentos aparte?
* [x] ¿Tienen coherencia suficiente las historias de usuario, y en caso
   de que no lo tengan, se ha escrito un *user journey* para
   aclararlo?
Sí, hay coherencia suficiente
* [x] ¿Todas las historias de usuario representan un beneficio para el
    mismo y se relacionan con la lógica de negocio? Es decir, ¿los usuarios
    *desean* hacerlo o tú deseas que el usuario te lo pida para hacerlo?
Sí, las historias de usuario presentan beneficios claros.

lmchaves committed:
Agrego los viajes de usuario

lmchaves committed:
Agrego los viajes de usuario

lmchaves committed:
Actualizo README

lmchaves committed:
Actualizo README

lmchaves committed:
Actualizo README

lmchaves committed:
Actualizo README

lmchaves committed:
Update user-journeys.md

lmchaves committed:
Agrego las historias de ususario

lmchaves committed:
Update user-stories.md

lmchaves committed:
Update README.md

lmchaves committed:
Update README.md

lmchaves committed:
Update README.md

lmchaves committed:
Update README.md

lmchaves committed:
Update README.md

lmchaves committed:
Update README.md

lmchaves committed:
Update user-stories.md

lmchaves committed:
Update user-journeys.md

lmchaves committed:
Update user-journeys.md

lmchaves committed:
Update user-stories.md

lmchaves committed:
Create milestones.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Update README.md

lmchaves committed:
Update README.md

lmchaves committed:
Update README.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Update milestones.md

GaelGoncalba commented:
Buenas, se me ha asignado revisar tu PR.

GaelGoncalba commented on docs/milestones.md:
Creo que los hitos del proyecto están bastante completos y siguen una progresión adecuada, simplemente sugeriría añadir en el documento las historias de usuario asociadas a cada milestone, o añadir un enlace para cada hito a su correspondiente sección en la pestaña Milestones de los Issues, para poder verlo todo más fácilmente.

lmchaves replied:
Corregido aqui: 5096c8b

lmchaves committed:
Agego la interacción del ususario con el sistema

lmchaves committed:
Agrego los enlaces a los hitos

lmchaves committed:
Agrego los enlaces a los hitos

lmchaves commented on docs/user-journeys.md:
Tienes razón en que detallar los pasos específicos que los usuarios seguirán para interactuar con el sistema puede ayudar a aclarar el proceso y minimizar cualquier duda en el futuro. 
Esta corregido aqui: da91d1b

GaelGoncalba reviewed (approved):
Por mi parte con esos cambios que has realizado creo que todo está correcto en el objetivo

lmchaves commented:
@JJ ha sido aprobado por un compañero, pero el resto que había asignado no han respondido nada, hice el pull request el miércoles pasado

lmchaves committed:
Elimino el README de la carpeta docs

Lo tenía desde un prinicpio en esta carpeta, por eso no decidí quietarlo.

lmchaves committed:
Cambio del Milestone 0

lmchaves commented on README.md:
```diff
@@ -2,8 +2,15 @@
-tiempos de entrega estimados de las piezas,
-priorización de las reparaciones según la urgencia del cliente y la disponibilidad de las piezas. Para ello, disponemos de los datos de costos de transporte y tiempos estimados de entrega entre las sedes, lo que nos permitirá optimizar estos procesos de manera eficiente y realizar pruebas sobre la mejora en la gestión del flujo de piezas.
 
+- Costos de transporte entre sedes.
```
Ahora hay alguna manera para que nos salgan estos cambios en este objetivo?

JJ replied:
Esto sigue aquí. Por favor, elíminalo.

lmchaves commented:
Si perdón, aqui estan los cambios: el del Readme 5ef0619 y del milestone 0 dc6ce92

lmchaves committed:
Update milestones.md

lmchaves committed:
Update milestones.md

lmchaves committed:
Actualizo el milestone 0

lmchaves commented:
Esta aquí corregido: 3c1c643 @JJ

lmchaves committed:
Aplicar cambios que no forman parte del objetivo actual

JJ commented:
> si haces cambios digas explícitamente en el mensaje de commit cómo has corregido el error indicado.

¿Crees que "update milestones.md" explica de alguna forma cómo has corregido los errores que te he indicado o que has visto en errores frecuentes?

lmchaves committed:
Explicación Cambios del commit 3c1c643

Claridad en el Lenguaje: Se ha eliminado la frase "modelo del problema", que anteriormente causaba confusión, y se ha definido claramente el enfoque en el **dominio del problema** relacionado con las Historias de Usuario HU1 y HU2. Esto responde al error frecuente de no entender qué se está modelizando, evitando la ambigüedad.

Eliminación de Términos Vagos: Se ha dejado de lado la frase "análisis de todos los HU" y se ha enfocado en un modelo inicial que represente las entidades y relaciones clave. Esto aborda el error de no especificar lo que se espera realmente, evitando términos que no añaden valor.

Reducción de Detalles Innecesarios: Se ha simplificado el texto para hacer que la documentación sea más clara y concisa, evitando la sobrecarga de información, en línea con la observación de que escribir mucho no necesariamente mejora la comprensión del concepto.

lmchaves commented:
En este nuevo commit: 96c82a4, he explicado los diferentes cambios del commit 3c1c643, que no había explicado anteriormente @JJ

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,27 @@
+# [ [M0] Milestone 0: Modelado del problema](https://github.com/lmchaves/OrganizarTaller/milestone/4)
+- Definir el dominio del problema centrado en las Historias de Usuario HU1 y HU2. Se desarrollará un modelo inicial que represente las entidades clave involucradas y las interacciones entre ellas, como la gestión del inventario, disponibilidad de piezas y transporte entre sedes.
+- __Entregable__:
+    - Código que implemente las entidades principales relacionadas con las HU1 y HU2, incluyendo: Piezas, Inventario, Sedes y Transporte.
```
Si ya sabes qué entidades hay, ¿para qué necesitas modelizarlo? Ya las tienes ahí.

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,27 @@
+- Definir el dominio del problema centrado en las Historias de Usuario HU1 y HU2. Se desarrollará un modelo inicial que represente las entidades clave involucradas y las interacciones entre ellas, como la gestión del inventario, disponibilidad de piezas y transporte entre sedes.
+- __Entregable__:
+    - Código que implemente las entidades principales relacionadas con las HU1 y HU2, incluyendo: Piezas, Inventario, Sedes y Transporte.
+    - El código debe mostrar las relaciones entre estas entidades, como la interacción entre inventario y sedes, el flujo de piezas entre sedes, y cómo los tiempos de transporte afectan la disponibilidad.
```
¿Estas "relaciones" son parte del dominio?

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,27 @@
+    - El código debe mostrar las relaciones entre estas entidades, como la interacción entre inventario y sedes, el flujo de piezas entre sedes, y cómo los tiempos de transporte afectan la disponibilidad.
+
+- __Viabilidad__:
+Se considerará completado cuando el código refleje correctamente las interacciones del dominio descrito en las HU1 y HU2.
```
Cuando una persona trabaje en tu objetivo 2 y te entregue un código, ¿cómo vas a comprobar estas interacciones del "dominio"?

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,27 @@
+- __Viabilidad__:
+Se considerará completado cuando Javier pueda realizar consultas de piezas, recibir tiempos estimados de entrega, y los tests automáticos validen la funcionalidad implementada.
+
+# [ [M2] Milestone 2:  Implementación de pedidos de piezas](https://github.com/lmchaves/OrganizarTaller/milestone/3)
```
¿Has leído la parte del [fichero de errores frecuentes](https://github.com/JJ/IV-/blob/master/errores/objetivo-1.md) donde pone:

> 

    En desarrollo ágil más allá de dos milestones por delante no tiene mucho sentido planificar. Pero si se hace, conviene que se mire la progresión de objetivos de la asignatura para que se ajusten, en lo posible, a lo que se encuentra ahí.

lmchaves replied:
No se muy bien, como cambiar este milestone, ya que trata sobre uno uno de los probelmas mas importantes, dentro de mi problema global, que es, que cuando pides una pieza, tienes que tener en cuenta en que sede está esa peiza, teniendo en cuenta el coste y tiempo.

Luego lo de prioridad sobre los clientes en un principio lo pensé para otro milestone.

JJ replied:
Si es tan importante, ¿por qué lo dejas para el 2?

lmchaves replied:
He organizado los milestones en esta secuencia para asegurar una progresión adecuada y la consulta de disponibilidad de piezas en el Milestone 1 es una funcionalidad más básica y manejable para iniciar el desarrollo y probar el sistema, mientras que el Milestone 2, que abarca la implementación de pedidos entre sedes con factores como costos de transporte y tiempos de entrega, es más complejo e introduce lógica adicional de optimización.

Pero si esto supone que la realización de los milestones es errónea, cambiaré el orden de dichos milestones.

JJ replied:
Sólo puedo repetirte lo que he puesto al principio. A partir de ahí, la secuencia debe responder a los intereses del cliente, y también al hecho de que si en el milestone 1 no se incluye una lógica de negocio suficiente no se podrá llevar a cabo el objetivo 4 de la asignatura.

JJ reviewed (commented):
En ningún momento te dije que "modelo del problema" era un error. Tanto "modelo del problema" como "dominio del problema" como "estructura del problema" son un error si no son consecuencia de entender lo que se está haciendo en este objetivo y para qué sirve un milestone/PMV sino un simple intento de "quitarse este objetivo de en medio"

Te aconsejaría que leyeras de nuevo [el guión](https://jj.github.io/IV/documentos/proyecto/1.Infraestructura#informaci%C3%B3n-adicional) y trates de entender de qué se trata aquí y por qué es importante que se haga correctamente. Te lo puedo explicar en clase, que no recuerdo haberlo hecho. En todas las sesiones hago énfasis en la importancia de asistir a clase, para entender las claves y la trascendencia de cada uno de los objetivo y qué es lo que se busca que aprendáis con cada uno de ellos.

lmchaves committed:
Eliminar duplicados en la descripción del problema

lmchaves committed:
Update milestones.md

Se ha corregido el Milestone 0, quitando las entidades que había escrito, ya que si las digo directamente no tiene sentido el modelado. 

Se ha explicado correctamente como realizar el dominio del problema (analizando los HUs correspondientes)

lmchaves commented:
He corregido el milestone 0: [c4c5b87](https://github.com/lmchaves/OrganizarTaller/pull/7/commits/c4c5b87ef65da287f528b19e33434223f03649c8) @JJ

JJ commented on README.md:
```diff
@@ -2,13 +2,21 @@
 
 Para ello, disponemos de los datos de costos de transporte y tiempos estimados de entrega entre las sedes, lo que nos permitirá optimizar estos procesos de manera eficiente y realizar pruebas sobre la mejora en la gestión del flujo de piezas.
+=======
+>>>>>>> 8141f43 (Aplicar cambios que no forman parte del objetivo actual)
```
Estás dejando marcas de conflicto de fusión.

JJ reviewed (changes requested):
Tienes que resolver todos los comentarios.

lmchaves committed:
Update milestones.md

Modificado el milestone 1 para incluir lógica de negocio suficiente y diferenciarlo del milestone 2. Ahora el milestone 1 se centra en la consulta de disponibilidad de piezas y tiempos estimados sin optimización, dejando la lógica avanzada de selección de sedes y priorización para el milestone 2.

lmchaves commented:
Revisión [d4656ff](https://github.com/lmchaves/OrganizarTaller/pull/7/commits/d4656ffdd1d1a80e8f1decdd6b5786c5480a0dde) @JJ

JJ commented on docs/README.md:
Ya os he repetido que sólo deben aparecer aquí los ficheros que se revisen. Si queréis borrar o cambiar algo, hacedlo en main, por favor, y luego rebase.

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,25 @@
+# [ [M0] Milestone 0: Modelado del problema](https://github.com/lmchaves/OrganizarTaller/milestone/4)
+- Hay que realizar un modelo inicial sobre el dominio del problema, explicado en las historias de usuario [HU1](https://github.com/lmchaves/OrganizarTaller/issues/4) y [HU2](https://github.com/lmchaves/OrganizarTaller/issues/5), incluyendo las entidades y las relaciones entre ellas, necesarias para conseguir el PMV.
```
¿Puedes poner un ejemplo de una "relación entre entidades"?
En todo caso, ¿esta frase añade información sobre el "modelado del problema" o el "modelo inicial *sobre* el dominio del problema"? ¿Las dos cosas son lo mismo? ¿Hay que modelar "el problema" o "sobre el dominio del problema"? ¿Las entidades y relaciones son exactamente qué? ¿El modelo inicial? ¿El dominio del problema? ¿Ninguna de las  dos cosas?

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,25 @@
+# [ [M0] Milestone 0: Modelado del problema](https://github.com/lmchaves/OrganizarTaller/milestone/4)
+- Hay que realizar un modelo inicial sobre el dominio del problema, explicado en las historias de usuario [HU1](https://github.com/lmchaves/OrganizarTaller/issues/4) y [HU2](https://github.com/lmchaves/OrganizarTaller/issues/5), incluyendo las entidades y las relaciones entre ellas, necesarias para conseguir el PMV.
+- __Entregable__:
+    - Código de las entidades definidas.
```
¿Qué entidades? ¿El PMV es "entidades"?

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,25 @@
+- __Entregable__:
+    - Código de las entidades definidas.
+- __Viabilidad__:
+Se considerará completado cuando el código refleje correctamente el dominio del problema descrito en las [HU1](https://github.com/lmchaves/OrganizarTaller/issues/4) y [HU2](https://github.com/lmchaves/OrganizarTaller/issues/5).
```
Entonces el "modelo inicial" ¿qué es? 
Por otro lado, ¿divides el problema en dos historias de usuario pero luego se trabaja en las dos a la vez? ¿Por qué razón?

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,25 @@
+Se considerará completado cuando el código refleje correctamente el dominio del problema descrito en las [HU1](https://github.com/lmchaves/OrganizarTaller/issues/4) y [HU2](https://github.com/lmchaves/OrganizarTaller/issues/5).
+
+# [ [M1] Milestone 1: Implementación de la consulta de disponibilidad de piezas](https://github.com/lmchaves/OrganizarTaller/milestone/2)
+- IImplementar la funcionalidad que permita a Javier, el mecánico, consultar la disponibilidad de piezas en el sistema. La información incluirá si la pieza está disponible en el taller local y, en caso de no estarlo, mostrará los tiempos estimados de entrega desde otras sedes.
```
Pone "II". 
¿Estás diciendo ya qué hay que implementar? ¿Para qué quieres las entidades entonces (o el modelo, o el modelo sobre el dominio?

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,25 @@
+- IImplementar la funcionalidad que permita a Javier, el mecánico, consultar la disponibilidad de piezas en el sistema. La información incluirá si la pieza está disponible en el taller local y, en caso de no estarlo, mostrará los tiempos estimados de entrega desde otras sedes.
+
+- __Entregable__:
+    - Código que implemente la consulta de disponibilidad de piezas en el inventario, junto con los tiempos estimados de entrega.
```
¿Ya sabes lo que va a implementar? ¿Para qué sirven las historias de usuario entonces?

JJ reviewed (changes requested):
Por favor, tratad de entender correctamente 
1. Qué es un PMV
2. Qué importancia tiene el primer PMV
3. Qué relación hay entre los PMVs y las historias de usuario.

Y, en general, para qué sirve todo esto y por qué diablos insisto en pediros esto en esta asignatura.

lmchaves committed:
Actualizo Readme

lmchaves committed:
Merge branch 'Objetivo-1'

JJ commented:
Has cerrado accidentamente el PR, ¿no?

lmchaves commented:
Si, accidentalmente, cerré el pull request asociado al objetivo1. Fue un error de manipulación mientras trabajaba en la rama, y ya estoy en proceso de solucionarlo para asegurar que los cambios necesarios puedan revisarse.

He creado un nuevo pull request para continuar con el flujo de revisión sin problemas. Aprecio su comprensión y lamento las molestias causadas.

Este es el enlace https://github.com/lmchaves/OrganizarTaller/pull/8/ , no se como cambiar en el archivo de enlaces, sin tener que realizar un pull request y se me asigne de nuevo 5 revisores.

JJ commented:
Igual no vendría mal pasar por esa segunda ronda de revisores. Lo que tú veas.

lmchaves commented:
Vale pues lo envío.
